security/oc_tls.c

```diff
@@ -529,25 +555,26 @@ oc_tls_handler_schedule_write(oc_tls_peer_t *peer)
 #endif /* OC_CLIENT */
 
 static oc_event_callback_retval_t
-oc_tls_inactive(void *data)
+oc_dtls_inactive(void *data)
 {
-  OC_DBG("oc_tls: DTLS inactivity callback");
   oc_tls_peer_t *peer = (oc_tls_peer_t *)data;
-  if (is_peer_active(peer)) {
-    oc_clock_time_t time = oc_clock_time();
-    time -= peer->timestamp;
-    if (time < (oc_clock_time_t)OC_DTLS_INACTIVITY_TIMEOUT *
-                 (oc_clock_time_t)OC_CLOCK_SECOND) {
-      OC_DBG("oc_tls: Resetting DTLS inactivity callback");
-      return OC_EVENT_CONTINUE;
-    }
+  OC_DBG("oc_tls: DTLS inactivity callback for peer(%p)", (void *)peer);
+  if (!is_peer_active(peer)) {
+    OC_DBG("oc_tls: Terminating DTLS inactivity callback");
+    return OC_EVENT_DONE;
+  }
+  oc_clock_time_t time = oc_clock_time_monotonic();
+  time -= peer->timestamp;
+  if (time < DTLS_INACTIVITY_TIMEOUT_TICKS) {
+    OC_DBG("oc_tls: Resetting DTLS inactivity callback");
+    return OC_EVENT_CONTINUE;
+  }
+  mbedtls_ssl_close_notify(&peer->ssl_ctx);
+  if ((peer->endpoint.flags & TCP) == 0) {
     mbedtls_ssl_close_notify(&peer->ssl_ctx);
```

Comment:
From this code it seems to be more general - not only for DTLS (udp tls), but also for TLS (TCP) connection